app/models/notification.rb

```diff
@@ -0,0 +1,7 @@
+class Notification < ActiveRecord::Base
+  belongs_to :submission
+  belongs_to :task
```

Comment:
Please delete notifications on task or submission delete too. You can try modifying `Task` and `Submission` models for this.